store/file/ods.go

```diff
@@ -0,0 +1,269 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	eds "github.com/celestiaorg/celestia-node/share/new_eds"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ eds.AccessorCloser = (*OdsFile)(nil)
+
+type OdsFile struct {
+	path string
+	hdr  *header
+	fl   *os.File
+
+	lock sync.RWMutex
+	ods  square
+}
+
+// OpenOdsFile opens an existing file. File has to be closed after usage.
+func OpenOdsFile(path string) (*OdsFile, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := readHeader(f)
+	if err != nil {
+		return nil, err
+	}
+
+	return &OdsFile{
+		path: path,
+		hdr:  h,
+		fl:   f,
+	}, nil
+}
+
+// CreateOdsFile creates a new file. File has to be closed after usage.
+func CreateOdsFile(
+	path string,
+	datahash share.DataHash,
+	eds *rsmt2d.ExtendedDataSquare,
+) (*OdsFile, error) {
+	f, err := os.Create(path)
+	if err != nil {
+		return nil, fmt.Errorf("file create: %w", err)
+	}
+
+	h := &header{
+		version:    fileV0,
+		fileType:   ods,
+		shareSize:  share.Size, // TODO: rsmt2d should expose this field
+		squareSize: uint16(eds.Width()),
+		datahash:   datahash,
+	}
+
+	err = writeOdsFile(f, h, eds)
+	if err != nil {
+		return nil, fmt.Errorf("writing ODS file: %w", err)
+	}
+
+	// TODO: fill ods field with data from eds
+	return &OdsFile{
+		path: path,
+		fl:   f,
+		hdr:  h,
+	}, f.Sync()
```

Comment:
I read this file 3 times and only on the last iteration I noticed this `f.Sync`, what about to move it right after `if` above?